### PR TITLE
Introduce prelease workflow

### PR DESCRIPTION
**Pull request by sventschui**
Introduce prelease workflow, no publish to npm yet but this can then be done using `npm publish <url to tar ball>`

### CODE REVIEW

File: .github/workflows/prelease.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      - name: assert on correct commit
+        run: |
+          if [[ "$(git rev-parse HEAD)" != "${{ github.event.client_payload.latest_commit_sha }}" ]]; then
+            echo "Branch ${{ github.event.client_payload.branch }} checked out with SHA $(git rev-parse HEAD) but expected ${{ github.event.client_payload.latest_commit_sha }}"
+            exit 1
+          fi
+
+      - name: git config
+        run: |
+          git config --global user.email "sventschui@gmail.com"
```
Comment (JoviDeCroock) [inline .github/workflows/prelease.yml:23]: let's make this the `hello@preact.dev` account

**Comment by sventschui** [comment]
Not sure if I understand correctly but isn‘t this what is available in the changelog editor at https://prelease.netlify.app ?

**Comment by marvinhagemeister** [comment]
@JoviDeCroock It formats in the same way as `changelogged`, so we're all good :tada:

File: .github/workflows/prelease.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      - run: npm version ${{ github.event.client_payload.version }}
+
+      - run: npm pack
+
+      - name: Set PRELEASE_GIT_TAG
+        run: echo "::set-env name=PRELEASE_GIT_TAG::$(cat package.json | jq -r '["v",.version]|join("")')"
+
+      - run: git push origin refs/tags/${{ env.PRELEASE_GIT_TAG }}
+
+      - run: git push origin ${{ github.event.client_payload.branch }}
```
Comment (marvinhagemeister) [inline .github/workflows/prelease.yml:35]: I'm wondering how this plays out with protected branches. Currently `master` is protected.

**Pull request closed without merging**